### PR TITLE
feat(knowledge-base): follow the convention

### PR DESCRIPTION
**Pull request by Yougigun**
Because 

of consistency with other services

This commit 

follows the convention

Note:
this PR is only for POC, and there are some known issues will be refactored soon.

### CODE REVIEW

File: artifact/artifact/v1alpha/artifact.proto [hunk 1]
```diff
@@ -144,10 +144,23 @@ message KnowledgeBase {
...
+  string owner = 7;
   // The knowledge base tags.
-  repeated string tags = 4;
+  repeated string tags = 8;
 }
 
+// KnowledgeBasesList represents a list of knowledge bases.
+message KnowledgeBasesList {
+  // The list of knowledge bases.
+  repeated KnowledgeBase knowledgebases = 1;
```
Comment (donch1989) [inline artifact/artifact/v1alpha/artifact.proto:162]:
```suggestion
  repeated KnowledgeBase knowledge_bases = 1;
```

File: artifact/artifact/v1alpha/artifact.proto [hunk 2]
```diff
@@ -144,10 +144,23 @@ message KnowledgeBase {
   string name = 2;
   // The knowledge base description.
   string description = 3;
+  // Whether the knowledge base is in use.
+  bool inuse = 4;
```
Comment (donch1989) [inline artifact/artifact/v1alpha/artifact.proto:148]:
```suggestion
  bool in_use = 4;
```

Can you share some use cases of `inuse`?

File: artifact/artifact/v1alpha/artifact.proto [hunk 3]
```diff
@@ -191,6 +203,10 @@ message UpdateKnowledgeBaseRequest {
   string description = 3;
   // The knowledge base tags.
   repeated string tags = 4;
+  // The knowledge base owner.
+  string owner = 5;
```
Comment (donch1989) [inline artifact/artifact/v1alpha/artifact.proto:207]:
```suggestion
  string owner_name = 5;
```

We should follow the style used in other [backends](https://github.com/instill-ai/protobufs/blob/6e9050ef9d10eb61cc000e4bf19ce6de95d404e0/vdp/pipeline/v1beta/pipeline.proto#L306). We use `owner_name` to indicate the path `users/<user_id>`, `organizations/<org_id>`, and we use `owner` to embed the owner data.

**Review by donch1989 — CHANGES_REQUESTED** [review]
1. Besides the above comments, for consistency, we should use ListKnowledgeBases and GetKnowledgeBase for our gRPC endpoint naming.
2. How do we handle getting KnowledgeBase from different owners?
3. There are some typos in the PR title and message.

File: artifact/artifact/v1alpha/artifact.proto (hunk 1 above)
Comment (Yougigun) [reply artifact/artifact/v1alpha/artifact.proto:162]: done

File: artifact/artifact/v1alpha/artifact.proto (hunk 2 above)
Comment (Yougigun) [reply artifact/artifact/v1alpha/artifact.proto:148]: it probably will be a controller or a indicator whether this kb is used moved out due to not be included in this scope.

File: artifact/artifact/v1alpha/artifact.proto (hunk 3 above)
Comment (Yougigun) [reply artifact/artifact/v1alpha/artifact.proto:207]: I see.  for now, I will change the first.

**Comment by donch1989** [comment]
Hi @Yougigun ,
There are still some design differences compared to other backends. Can you refer to the pipeline or model API design as an example? Keeping the same API design will help clients integrate more easily.

**Comment by Yougigun** [comment]
> Hi @Yougigun , There are still some design differences compared to other backends. Can you refer to the pipeline or model API design as an example? Keeping the same API design will help clients integrate more easily.

Thank you for pointing out.
That's right. I had not noticed the design consistency with other services. Due to mvp testing purposes, I will refactor it in the next version(beta) with @thewbuk

**Comment by donch1989** [comment]
Hi, @Yougigun 
Please write a note in commit message to show that this PR is only for POC, and there are some known issues will be refactored soon.

**Comment by Yougigun** [comment]
> Hi, @Yougigun Please write a note in commit message to show that this PR is only for POC, and there are some known issues will be refactored soon.

done.

**Review by donch1989 — APPROVED** [review]

**Pull request merged**